imap_processing/swe/l1b/swe_l1b_science.py

```diff
@@ -254,21 +205,31 @@ def apply_in_flight_calibration(
     Parameters
     ----------
     corrected_counts : numpy.ndarray
-        Corrected count of quarter cycle data. Data shape is (180, 7).
-    acquisition_time : float
-        Acquisition time of current quarter cycle data.
+        Corrected count of full cycle data. Data shape is (24, 30, 7).
+    acquisition_time : numpy.ndarray
+        Acquisition time of full cycle data. Data shape is (24, 30, 7).
 
     Returns
     -------
     corrected_counts : numpy.ndarray
-        Corrected count of quarter cycle data after applying in-flight calibration.
-        Array shape is (180, 7).
+        Corrected count of full cycle data after applying in-flight calibration.
+        Array shape is (24, 30, 7).
     """
-    # calculate calibration factor
-    cal_factor = calculate_calibration_factor(acquisition_time)
-    # Apply to current quarter cycle data
+    # Read in in-flight calibration data
+    in_flight_cal_df = read_in_flight_cal_data()
+    # calculate calibration factor.
+    # return shape of interp_func is (24, 30, 7, 7) where last 7 dimension contains
+    # calibration factor for each CEM detector.
+    cal_factor = interp_func(
+        acquisition_time,
```

Comment:
I assume the acquisition time for each CEM detector is the same? Otherwise I don't think the for loop below would work as expected since it is picking off the diagonal. I'd suggestion just taking the first acquisition time, or only passing in the shape (24, 30) to this function.
```suggestion
        acquisition_time[..., 0],
```